dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,400 @@
+                    "Invalid attempt to create a service descriptors performed by user with id `%s`.",
+                    user.getUserId()));
+        }
+
+        if (getServiceDescriptorMap(user).containsKey(serviceKey)) {
+            throw new DotDataException(
+                    String.format("There's a service already registered under key '%s'.",
+                            serviceKey));
+        }
+
+        final String ymlFilesPath = getServiceDescriptorDirectory();
+        final File basePath = new File(ymlFilesPath);
+        if (!basePath.exists()) {
+            basePath.mkdir();
+        }
+        Logger.debug(ServiceIntegrationAPIImpl.class,
+                () -> " ymlFiles are set under:  " + ymlFilesPath);
+        final ServiceDescriptor serviceDescriptor = ymlMapper
+                .readValue(inputStream, ServiceDescriptor.class);
+        ymlMapper.writeValue(new File(basePath, String.format("%s.yml", serviceKey)),
+                serviceDescriptor);
+
+        CacheLocator
+                .getCacheAdministrator()
+                .remove(DESCRIPTORS_LIST_KEY, DESCRIPTORS_CACHE_GROUP);
+        CacheLocator.getCacheAdministrator()
+                .remove(DESCRIPTORS_MAPPED_BY_SERVICE_KEY, DESCRIPTORS_CACHE_GROUP);
+
+    }
+
+
+    private static String getServiceDescriptorDirectory() {
+        final Supplier<String> supplier = () -> APILocator.getFileAssetAPI().getRealAssetsRootPath()
+                + File.separator + "server" + File.separator;
+        final String dirPath = Config
+                .getStringProperty(SERVICE_INTEGRATION_DIR_PATH_KEY, supplier.get());
+        return Paths.get(dirPath).normalize().toString();
+    }
+
+    private Set<String> listAvailableYamlFiles() throws IOException {
+        final String ymlFilesPath = getServiceDescriptorDirectory();
+        final File basePath = new File(ymlFilesPath);
+        if (!basePath.exists()) {
+            basePath.mkdir();
+        }
+        Logger.debug(ServiceIntegrationAPIImpl.class,
+                () -> " ymlFiles are set under:  " + ymlFilesPath);
+        final Set<String> files = listFiles(ymlFilesPath);
+        if (!UtilMethods.isSet(files)) {
+            // No files were found a default empty descriptor will be created.
+            ymlMapper.writeValue(new File(basePath, "sample-service-descriptor.yml"),
+                    emptyDescriptor());
+            return listFiles(ymlFilesPath);
+        } else {
+            return files;
+        }
+    }
+
+    private Set<String> listFiles(final String dir) throws IOException {
+        final Set<String> fileList = new HashSet<>();
+        try (final DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(dir))) {
+            for (final Path path : stream) {
+                if (!Files.isDirectory(path)) {
+                    final String fileName = path.getFileName().toString();
+                    if (fileName.equals("yaml") || fileName.endsWith("yml")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)